### PR TITLE
fix mpu9250 for edison build

### PR DESCRIPTION
**Pull request by AD-Aleksandrov**

### CODE REVIEW

**Comment by julianoes** [comment]
Not sure why the travis link disappeared but it's doing it:
https://travis-ci.org/PX4/DriverFramework/builds/144496363

**Comment by julianoes** [comment]
Oh, I pressed restart, and now it re-appeared here :).

**Comment by julianoes** [comment]
Aha, now it asks you to `make fix-style` :).

**Comment by AD-Aleksandrov** [comment]
I have done it:)

File: drivers/mpu9250/MPU9250.cpp [hunk 1]
```diff
@@ -160,8 +160,15 @@ int MPU9250::mpu9250_init()
 	//	DF_LOG_ERR("sample rate config failed");
 	//}
 	//usleep(1000);
+
+#if defined(__EDISON)
+	//FIFO corrupt and overflow at 250Hz.
+	result = _writeReg(MPUREG_CONFIG,
+			   BITS_DLPF_CFG_184HZ | BITS_CONFIG_FIFO_MODE_OVERWRITE);
```
Comment (julianoes) [inline drivers/mpu9250/MPU9250.cpp]: This hints that something is not fast enough to grab the samples? Any idea why you experience this? Could anything else be slowing it down?

Comment (AD-Aleksandrov) [inline drivers/mpu9250/MPU9250.cpp]: Don't you have the same problem on your device? The temperature sensor samples at 4000Hz (using DLPF_CFG = 0) and it is faster then frequency of _measure, or i don't understand something?

Comment (julianoes) [inline drivers/mpu9250/MPU9250.cpp]: That's why it used the FIFO buffer of the device. The current driver is actually running at 1000 Hz (see: https://github.com/PX4/DriverFramework/blob/master/drivers/mpu9250/MPU9250.hpp#L182), so it collects 8 samples each time. 

There were some problems when the mag was added but they have recently been fixed, see #117.

Comment (AD-Aleksandrov) [inline drivers/mpu9250/MPU9250.cpp]: TEST Verify Schedule: FAILED
Any idea why? Another processes may slow down driver?

Comment (julianoes) [inline drivers/mpu9250/MPU9250.cpp]: Yes, it's about the scheduling on your device. You can try with CPU governors set to performance.

Comment (AD-Aleksandrov) [inline drivers/mpu9250/MPU9250.cpp]: performance has set

Comment (julianoes) [inline drivers/mpu9250/MPU9250.cpp]: Can you paste the whole output? And maybe create a separate issue with it?

**Comment by julianoes** [comment]
@AD-Aleksandrov I think it would be worthwhile to test again on Edison using the default config. Can you please post the whole output and then we can try to re-asses the situation of which sample frequency makes sense to use?

**Comment by AD-Aleksandrov** [comment]
Yes, i will have posted by monday. Thank you.

**Comment by AD-Aleksandrov** [comment]
[test_mpu.txt](https://github.com/PX4/DriverFramework/files/381339/test_mpu.txt)

mpu config - BITS_DLPF_CFG_184HZ (1000kHz),
and SPI speed - 5MHz.
This config works without overflow, corrupt fifo.

**Comment by julianoes** [comment]
@AD-Aleksandrov: so 8000kHz is not possible at 1MHz or 5Mhz?

The problem is that setting the DLPF_CFG to 184Hz will need some changes in the update rate and publishing logic as well, I think.

**Comment by AD-Aleksandrov** [comment]
8kHz is possible at 5 and 1Mhz, but fifo corrupts many times at 10Mhz.
5Mhz spi speed is the best value, I think.

https://github.com/PX4/DriverFramework/blob/77cb39a3e3383d6be41fb5bce5f265301d557111/drivers/mpu9250/MPU9250.hpp#L235
Is it possible to add 1.0f?

**Comment by julianoes** [comment]
Yes that's possible. I would like to keep the diff as small as possible and I'd therefore vote for 8kHz at 5Mhz. Why do you want to use the DLPF at 184Hz?

**Comment by AD-Aleksandrov** [comment]
There are two problems: 
- fifo overflow at 8kHz,
- fifo does not overflow at 1kHz but sometimes corrupt at 10MHz.
config - 8kHz and 10MHz - fifo overflow and corrupt,
config - 1kHz and 10MHz - fifo corrupt.
Driver cannot work on edison at 1kMHz(MPU9250_MEASURE_INTERVAL_US) so we need to decrease sample rate, and there is a problem with spi speed. So we need to change both parameters.

**Comment by julianoes** [comment]
Understood, I'd say go ahead then to adapt it to your needs with ifdefs including the change at https://github.com/PX4/DriverFramework/blob/77cb39a3e3383d6be41fb5bce5f265301d557111/drivers/mpu9250/MPU9250.hpp#L235.

**Comment by julianoes** [comment]
Sorry if you misunderstood me. I meant "make all the ifdef changes" so that we can merge it.

**Comment by AD-Aleksandrov** [comment]
Yes, I will add it. Thanks.

**Pull request closed without merging**